### PR TITLE
examples: Add rgw script to access admin APIs using curl

### PR DESCRIPTION
**Pull request by pdvian**
The rgw_admin_curl.sh script will allow end-user/developers to
access RGW admin APIs through curl command.

Signed-off-by: Prashant D <pdhange@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by dang — APPROVED** [review]

File: examples/rgw_admin_curl.sh [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Library Public License as published by
+# the Free Software Foundation; either version 2, or (at your option)
+# any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Library Public License for more details.
```
Comment (cbodley) [inline examples/rgw_admin_curl.sh:15]: is it really necessary to use a GPL license here? ceph as a whole is LGPL, so this script would be covered by default. but mixing GPL and LGPL just complicates things unnecessarily

File: examples/rgw_admin_curl.sh [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+${dateTime}
+${resource}"
+
+signature=`echo -en "$headerToSign" | \
+           openssl sha1 -hmac ${secret_key} -binary | base64`
+
+echo "Sending ${http_request} request to ${rgw_endpoint} :" 
+curl -X ${http_request} -H "Content-Type: ${contentType}" -H "Date: ${dateTime}" \
+     -H "Authorization: AWS ${access_key}:${signature}" -H "Host: ${rgw_endpoint}" \
+     "http://${rgw_endpoint}${resource}?${http_query}" 2> /dev/null|jq "."
```
Comment (cbodley) [inline examples/rgw_admin_curl.sh:92]: i feel like this script would be more flexible if it didn't pipe to `jq .`. users may not have jq installed, or they may want to do their own filtering on top

to support that, we'd want to remove the `echo`s too. what do you think?

**Review by cbodley — COMMENTED** [review]
thanks @pdvian!

**Comment by alimaredia** [comment]
@pdvian From when you showed us the script earlier you had these examples you sent us:
To create rgw user:
`$ sh rgw_curl.sh "NU1PBZEYX0Q81CBQ979O" "7E5TTkTDEueWNMTywdiXetQkp017YdASgMIJ2ezd" 10.0.0.0:8080 PUT user "uid=test123&display-name=usertest123"`

To get rgw user info:
`$ sh rgw_curl.sh "NU1PBZEYX0Q81CBQ979O" "7E5TTkTDEueWNMTywdiXetQkp017YdASgMIJ2ezd" 10.0.0.0:8080 GET user "uid=test123"`

To list buckets:
`$ sh rgw_curl.sh "NU1PBZEYX0Q81CBQ979O" "7E5TTkTDEueWNMTywdiXetQkp017YdASgMIJ2ezd" 10.0.0.0:8080 GET bucket ""`

Could you add those to the script somehow? Maybe in the help section? They were very useful when I used the script for the first time.

File: examples/rgw_admin_curl.sh (hunk 1 above)
Comment (pdvian) [reply examples/rgw_admin_curl.sh:15]: Ack. I will remove the GPL license.

File: examples/rgw_admin_curl.sh (hunk 2 above)
Comment (pdvian) [reply examples/rgw_admin_curl.sh:92]: Yes. Makes sense. Let me remove jq part from default behavior and add option to use jq if specified.  Removing `echo` s too.

**Review by cbodley — APPROVED** [review]

**Comment by cbodley** [comment]
tyvm @pdvian!

**Pull request merged**